### ISSUE TITLE
mypy_path is being ignored when providing file arguments

### ISSUE THREAD

**Issue by twoertwein** [python/mypy#13098]
**Bug Report**

Mypy seems to ignore the value of mypy_path when file argument are provided.

**To Reproduce**
Consider a folder structure like this:
```
- awesome-stubs
- awesome-stubs/__init__.py
- tests
- tests/test.py
- pyproject.toml
```
with
```sh
$ cat awesome-stubs/__init__.py 
def super_awesome() -> int:
    return 1
```

and 

```sh
$ cat tests/test.py 
import awesome

reveal_type(awesome.super_awesome())
```
and
```sh
$ cat pyproject.toml
[tool.mypy]
mypy_path = "."
files = ["awesome-stubs", "tests"]
```
**Expected Behavior**

Expected behavior (when providing no file arguments)
```sh
$ mypy
tests/test.py:3: note: Revealed type is "builtins.int"
Success: no issues found in 2 source files
```
Expected
```sh
$ mypy tests/test.py
tests/test.py:3: note: Revealed type is "builtins.int"
Success: no issues found in 2 source files
```

**Actual Behavior**

```sh
$ mypy tests/test.py
tests/test.py:1: error: Cannot find implementation or library stub for module named "awesome"
tests/test.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
tests/test.py:3: note: Revealed type is "Any"
Found 1 error in 1 file (checked 1 source file)

```

**Your Environment**

- Mypy version used: 0.961
- Python version used: 3.10
- Operating system and version: Fedora
